### ISSUE TITLE
[>=0.6.5] ImportError when using an installed package having Cython extension

### ISSUE THREAD

**Issue by AxelVoitier** [Nuitka/Nuitka#610]
Hello,

I noticed the following regression:
From 0.6.5 and above (including develop branch), compiling a script (with --recurse-all) using a pip-installed packaged having a Cython extension, running the produced executable lead to ImportError.

Here is a self contained reproducible test, reduced to minimum setup:
https://github.com/AxelVoitier/nuitka-cython-test
Here is the content of the used Dockerfile for info on how it is setup:
```

FROM python:3.7.6-slim-buster

RUN apt-get update
RUN apt-get install -y --no-install-recommends gcc g++
RUN pip --no-cache-dir install Cython==0.29.14

# RUN pip --no-cache-dir install nuitka==0.6.4  # Working
RUN pip --no-cache-dir install nuitka==0.6.7  # ImportError
# RUN pip --no-cache-dir install https://github.com/Nuitka/Nuitka/archive/develop.zip  # ImportError

COPY fib-package/ /app/fib-package/
WORKDIR /app

RUN pip install fib-package/
RUN nuitka3 --show-progress --recurse-all /usr/local/bin/fib

CMD ["/app/fib.bin"]
```

In my real use case, I don't directly use the installed package. It's rather a dependency of one or more packages. Then, I nuitka the entire final app. The SSCCE in the repo reproduce enough of that setup though.

Using nuitka 0.6.4 gives:
```
docker run -it --rm axelvoitier/nuitka-cython-test
1 1 2 3 5 8
```

Using nuitka 0.6.5 and above gives:
```
docker run -it --rm axelvoitier/nuitka-cython-test
Traceback (most recent call last):
  File "/usr/local/bin/fib", line 5, in <module>
    from fib.main import main_fn
  File "/usr/local/lib/python3.7/site-packages/fib/main.py", line 1, in <module fib.main>
    from fib import fib_ext
ImportError: cannot import name 'fib_ext' from 'fib' (/usr/local/lib/python3.7/site-packages/fib/__init__.py)
Makefile:5: recipe for target 'run' failed
```

Checking that the lib is correctly installed:
```
$ docker run -it --rm axelvoitier/nuitka-cython-test ls -la /usr/local/lib/python3.7/site-packages/fib
total 108
drwxr-xr-x 3 root root  4096 Jan 28 11:23 .
drwxr-xr-x 1 root root  4096 Jan 28 11:23 ..
-rw-r--r-- 1 root root     0 Jan 28 11:23 __init__.py
drwxr-xr-x 2 root root  4096 Jan 28 11:23 __pycache__
-rwxr-xr-x 1 root root 93888 Jan 28 11:23 fib_ext.cpython-37m-x86_64-linux-gnu.so
-rw-r--r-- 1 root root   104 Jan 28 11:23 main.py
```

I could not find a corresponding issue here. Although I found similar issues with "well known" packages like Panda having hidden imports. I don't think the problem here is hidden import though.

Cheers,
Axel

**Comment by AxelVoitier** [comment]
Also, just Cythonizing locally and then compilling with nuitka the bare main script works (ie. bypassing entirely the package installation).

Could that mean the issue could come from search paths maybe?

**Comment by AxelVoitier** [comment]
Hello,

By dichotomic search in the git commits, I found out this commit is the one that introduced the bug:
https://github.com/Nuitka/Nuitka/commit/eb9b32a509106dd4b8676c06fac665a1b552b141


Cheers,
Axel

**Comment by AxelVoitier** [comment]
Hello,

@kayhayen, any insight? Anything I can do to make it easier for you to figure it out?
Can this at least get triaged?


Cheers,
Axel

**Comment by AxelVoitier** [comment]
Fixed by a5490c4e3536e0f8c3c8004f2b5e91bd47f3ad8f.


Cheers,
Axel

**Comment by kayhayen** [comment]
Sorry, this issue went by the radar somehow, but apparently it was a duplicate anyway.
